engine/apps/mattermost/tests/test_tasks.py

```diff
@@ -189,3 +214,290 @@ def test_on_alert_group_action_triggered_async_failure(
+    make_alert_receive_channel,
+    make_alert_group,
+    make_alert,
+    make_mattermost_channel,
+    make_mattermost_post_response,
+    make_mattermost_message,
+    make_user_notification_policy,
+    make_mattermost_user,
+):
+    organization, user = make_organization_and_user()
+
+    alert_receive_channel = make_alert_receive_channel(organization)
+    alert_group = make_alert_group(alert_receive_channel)
+    make_alert(alert_group=alert_group, raw_request_data=alert_receive_channel.config.example_payload)
+
+    make_mattermost_channel(organization=organization, is_default_channel=True)
+    make_mattermost_message(alert_group, MattermostMessage.ALERT_GROUP_MESSAGE)
+
+    make_mattermost_user(user=user)
+    user_notification_policy = make_user_notification_policy(
+        user=user,
+        step=UserNotificationPolicy.Step.NOTIFY,
+        notify_by=UserNotificationPolicy.NotificationChannel.TESTONLY,
+    )
+
+    url = "{}/api/v4/posts".format(settings.MATTERMOST_HOST)
+    data = make_mattermost_post_response()
+    mock_response = httpretty.Response(json.dumps(data), status=status.HTTP_200_OK)
+    httpretty.register_uri(httpretty.POST, url, responses=[mock_response])
+
+    notify_user_about_alert_async(
+        user_pk=user.pk, alert_group_pk=alert_group.pk, notification_policy_pk=user_notification_policy.pk
+    )
+
+    mattermost_message = alert_group.mattermost_messages.order_by("created_at").last()
+    assert mattermost_message.post_id == data["id"]
+    assert mattermost_message.channel_id == data["channel_id"]
+    assert mattermost_message.message_type == MattermostMessage.USER_NOTIFACTION_MESSAGE
+
+
+@pytest.mark.django_db
+def test_notify_user_about_alert_async_user_does_not_exist(
+    make_organization_and_user,
+    make_alert_receive_channel,
+    make_alert_group,
+    make_alert,
+    make_mattermost_channel,
+    make_mattermost_message,
+    make_mattermost_user,
+    make_user_notification_policy,
+):
+    organization, user = make_organization_and_user()
+
+    alert_receive_channel = make_alert_receive_channel(organization)
+    alert_group = make_alert_group(alert_receive_channel)
+    make_alert(alert_group=alert_group, raw_request_data=alert_receive_channel.config.example_payload)
+    make_mattermost_channel(organization=organization, is_default_channel=True)
+    make_mattermost_message(alert_group, MattermostMessage.ALERT_GROUP_MESSAGE)
+    make_mattermost_user(user=user)
+    user_notification_policy = make_user_notification_policy(
+        user=user,
+        step=UserNotificationPolicy.Step.NOTIFY,
+        notify_by=UserNotificationPolicy.NotificationChannel.TESTONLY,
+    )
+
+    with patch("apps.mattermost.client.MattermostClient.create_post") as mock_post_call:
+        notify_user_about_alert_async(
+            user_pk=123, alert_group_pk=alert_group.pk, notification_policy_pk=user_notification_policy.pk
+        )
+
+    mock_post_call.assert_not_called()
```

Comment:
This should check for the failed log record (similar below).

Reply:
Since this is to check if the actual user present we cannot create log records as the user is required for creating the same. Hence I am just adding a log and returning from task.